pkg/resource/deploy/builtins.go

```diff
@@ -230,3 +239,23 @@ func (p *builtinProvider) readStackResourceOutputs(inputs resource.PropertyMap)
 		"outputs": resource.NewObjectProperty(outputs),
 	}, nil
 }
+
+func (p *builtinProvider) readStackResource(inputs resource.PropertyMap) (resource.PropertyMap, error) {
+	urn, ok := inputs["urn"]
+	if !ok {
+		return nil, fmt.Errorf("readStackResource missing required 'urn' argument")
+	}
+	if !urn.IsString() {
+		return nil, fmt.Errorf("readStackResource 'urn' argument expected string got %s", urn.TypeString())
+	}
+
+	state := p.plan.current[resource.URN(urn.StringValue())]
+	if state == nil {
+		return nil, fmt.Errorf("resource '%s' not yet registered in current stack state", urn.StringValue())
```

Comment:
would this not indicate a bug in the pulumi engine (or sdks) itself?  i.e. users get urns from existing resources, so if they called in with one and didn't find a resource, then we somehow gave them an invalid urn...

(or could this actually be  bug in their code if they manually constructed one themselves.  not sure if we allow for that anywhere).

Reply:
No - a user could just call this invoke any time, and they may call it at a time that it is invalid (the resource they are asking for has not been registered).  It really is up to users to ensure they only call this after then know the resource they are asking for has been registered (most likely by some other process).